subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -608,6 +618,73 @@ static void aws_fota_cb_handler(struct aws_fota_event *fota_evt)
 }
 #endif /* defined(CONFIG_AWS_FOTA) */
 
+static int nct_settings_set(const char *key, size_t len_rd,
```

Comment:
I'm very confused by the settings API.. This is a function for reading the settings, right, but it's registered as a `.h_set` handler?

Reply:
yes i don't know why it is like that either 🤷‍♂️ 

[h_set](https://docs.zephyrproject.org/latest/reference/settings/index.html?highlight=settings_subsys_init#handlers)  - This gets called when the value is loaded from persisted storage with **settings_load()**, or when using settings_runtime_set() from the runtime backend.